lib/cocoapods-core/master_source.rb

```diff
@@ -0,0 +1,38 @@
+module Pod
+  class MasterSource < Source
+    # @!group Updating the source
+    #-------------------------------------------------------------------------#
+
+    # Updates the local clone of the source repo.
+    #
+    # @param  [Bool] show_output
+    #
+    # @return  [Array<String>] changed_spec_paths
+    #          Returns the list of changed spec paths.
+    #
+    def update(show_output)
+      if requires_update
+        super
+      else
+        []
+      end
+    end
+
+    # Returns whether a source requires updating.
+    #
+    # @param [Source] source
+    #        The source to check.
+    #
+    # @return [Bool] Whether the given source should be updated.
+    #
+    def requires_update
+      current_commit_hash = '""'
+      Dir.chdir(repo) do
+        current_commit_hash = "\"#{(`git rev-parse HEAD`).strip}\""
```

Comment:
this should not be quoted, and getting the current commit hash should be extracted into a new method